pkg/controller/machineset.go

```diff
@@ -349,9 +355,20 @@ func (c *controller) manageReplicas(allMachines []*v1alpha1.Machine, machineSet
 		}
 
 		diff *= -1
+
 		if diff > BurstReplicas {
 			diff = BurstReplicas
 		}
+
+		// throttle machine creation until all stale machines are deleted
+		if len(staleMachines) > 0 && diff > machineDeletionWindow {
+			diff = machineDeletionWindow
+		}
```

Comment:
Thanks. This and other suggestions should be encoded into unit tests. I'll get right to it.

Reply:
Agreed, we should recompute the `diff` , with `diff = diff - (len(staleMachines) - len(machineDeletionWindow))` to avoid throttled machine-creations. 
This is clearly a bit mind-boggling, thanks for @ggaurav10 for catching and @zuzzas for implementing.

Reply:
It is done.
https://github.com/gardener/machine-controller-manager/pull/482/files#diff-16f7b87f40829e0c81cb6c486f3997ccR345-R359

Reply:
thanks. @ggaurav10 can you please take a quick look at this aspect.

Reply:
I see, now we are recomputing the diff at the start itself, and the new diff[which considers the left out stale machines]. At the first sight, it looks good, I'd like to test this aspect separately.